plugins/editorconfig/src/org/editorconfig/language/services/EditorConfigOptionDescriptorManager.kt

```diff
@@ -0,0 +1,26 @@
+// Copyright 2000-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+package org.editorconfig.language.services
+
+import com.intellij.openapi.components.ServiceManager
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.startup.StartupActivity
+import org.editorconfig.language.schema.descriptors.impl.EditorConfigDeclarationDescriptor
+import org.editorconfig.language.schema.descriptors.impl.EditorConfigOptionDescriptor
+
+interface EditorConfigOptionDescriptorManager {
+  fun getOptionDescriptor(parts: List<String>, smart: Boolean): EditorConfigOptionDescriptor?
+  fun getOptionDescriptors(smart: Boolean): Iterable<EditorConfigOptionDescriptor>
+  fun getDeclarationDescriptors(descriptor: EditorConfigDeclarationDescriptor): Iterable<EditorConfigDeclarationDescriptor>
+  fun getRequiredDeclarationDescriptors(descriptor: EditorConfigDeclarationDescriptor): Iterable<EditorConfigDeclarationDescriptor>
+
+  companion object {
+    val instance: EditorConfigOptionDescriptorManager
+      get() = ServiceManager.getService(EditorConfigOptionDescriptorManager::class.java)
+  }
+
+  class MyStartupActivity : StartupActivity {
```

Comment:
It would not be possible to abandon all of them, but this one should be ok as a service - I don't see anything that requires immediate instantiation. And yes, a component will be better if we need to do something before project is initialized